### PR TITLE
Update the top-level crate doc

### PR DESCRIPTION
**Pull request by jeehoonkang**
I updated the crate's top-level documentation to better reflect the current status.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
+//!   use, and high-performance APIs for managing memory in these cases. We plan
+//!   to support other memory management schemes, e.g. hazard pointers (HP) and
+//!   quiescent state-based reclamation (QSBR). The crate is reexported as the
+//!   `epoch` module.
 //!
-//! - **Memory management**. Because non-blocking data structures avoid global
-//! synchronization, it is not easy to tell when internal data can be safely
-//! freed. The `mem` module provides generic, easy to use, and high-performance
-//! APIs for managing memory in these cases. These live in the `mem` module.
+//! - `crossbeam-utils` for **Utilities**. A "scoped" thread API makes it
```
Comment (Vtec234) [inline src/lib.rs]: A "scoped" -> The "scoped"

File: src/lib.rs [hunk 2]
```diff
@@ -1,26 +1,29 @@
...
+//!   possible to spawn threads that share stack data with their
+//!   parents. `CachePadded` struct inserts pad to align data with
```
Comment (Vtec234) [inline src/lib.rs]: `CachePadded` struct inserts pad -> The `CachePadded` struct inserts padding

File: src/lib.rs [hunk 3]
```diff
@@ -1,26 +1,29 @@
...
+//!   cacheline. This crate also seeks to expand the standard library's a few
```
Comment (Vtec234) [inline src/lib.rs]: cacheline -> the size of a cacheline; a few -> few

File: src/lib.rs [hunk 4]
```diff
@@ -1,26 +1,29 @@
...
+//!   synchronization primitives (locks, barriers, etc) to include
+//!   advanced/niche primitives, as well as userspace alternatives. This crate
+//!   is reexported as the `utils` module. `CachePadded` and scoped thread API
+//!   is also reexported at the top-level.
```
Comment (Vtec234) [inline src/lib.rs]: is also -> are also

**Comment by jeehoonkang** [comment]
@Vtec234 Thanks!  I just updated this PR as you said.

**Comment by jeehoonkang** [comment]
Thanks :)  merging it!

File: README.md [hunk 5]
```diff
@@ -1,46 +1,43 @@
...
+   bags, sets and maps. These subcrates are reexported in the `sync` module.
 
 # Usage
 
 To use Crossbeam, add this to your `Cargo.toml`:
 
 ```toml
 [dependencies]
-crossbeam = "0.3.0"
+crossbeam = "0.4.0"
```
Comment (Lucretiel) [inline README.md]: crossbeam isn't in version 0.4.0? Latest I'm seeing on crates.io is 0.3.2

**Pull request merged**